stelaval/assignment4/IslandsCounterTest.java

```diff
@@ -0,0 +1,64 @@
+package assignment4;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+public class IslandsCounterTest {
+  
+  @Test
+  public void countIslandsExample() {
+    IslandsCounter counter = new IslandsCounter();
+    
+    boolean[][] map = new boolean[][] {
+      {false, true, false, true},
+      {true, true, false, false},
+      {false, false, true, false},
+      {false, false, true, false}
+    };
+    
+    assertEquals (counter.countIslands(map), 3);

```

Comment:
`assertEquals` assumes that expected value is passed as the first parameter and the actual value as the second. If you pass them the other way round, tests will still fail in case of error, but the error message will be confusing since it will mention incorrect value as expected one.